src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -1222,24 +1222,47 @@ bool LibraryCallKit::inline_vector_mem_masked_operation(bool is_store) {
 
   int mem_num_elem = using_byte_array ? num_elem * type2aelembytes(elem_bt) : num_elem;
   BasicType mem_elem_bt = using_byte_array ? T_BYTE : elem_bt;
-  bool use_predicate = arch_supports_vector(is_store ? Op_StoreVectorMasked : Op_LoadVectorMasked,
-                                            mem_num_elem, mem_elem_bt,
-                                            (VectorMaskUseType) (VecMaskUseLoad | VecMaskUsePred));
-  // Masked vector store operation needs the architecture predicate feature. We need to check
-  // whether the predicated vector operation is supported by backend.
-  if (is_store && !use_predicate) {
-    if (C->print_intrinsics()) {
-      tty->print_cr("  ** not supported: op=storeMasked vlen=%d etype=%s using_byte_array=%d",
-                    num_elem, type2name(elem_bt), using_byte_array ? 1 : 0);
+  bool supports_predicate = arch_supports_vector(is_store ? Op_StoreVectorMasked : Op_LoadVectorMasked,
+                                                mem_num_elem, mem_elem_bt,
+                                                (VectorMaskUseType) (VecMaskUseLoad | VecMaskUsePred));
+
+  // If current arch does not support the predicated operations, we have to bail
+  // out when current case uses the predicate feature.
+  if (!supports_predicate) {
+    bool needs_predicate = false;
+    if (is_store) {
+      // Masked vector store always uses the predicated store.
+      needs_predicate = true;
+    } else {
+      // Masked vector load with IOOBE always uses the predicated load.
+      const TypeInt* offset_in_range = gvn().type(argument(8))->isa_int();
```

Comment:
Should if be `argument(7)`? (and adjustments later to access the container).

Reply:
I'm afraid it's `argument(8)` for the load operation since the `argument(7)` is the mask input. It seems the argument number is not right begin from the mask input which is expected to be `6`. But the it's not. Actually I don't quite understand why.

Reply:
offset is long so uses two argument slots (5 and 6). 
mask is argument (7).
offsetInRange is argument(8).

Reply:
Make sense! Thanks for the explanation!

Reply:
Doh! of course. This is not the first and will not be the last time i get caught out by the 2-slot requirement.
It may be useful to do this:
```
Node* mask_arg = is_store ? argument(8) : argument(7);
```

Reply:
Yes, the mask argument is got like:
```
 Node* mask = unbox_vector(is_store ? argument(8) : argument(7), mbox_type, elem_bt, num_elem);
```